sys/riot_hdr/riot_hdr.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_riot_hdr
+ * @{
+ *
+ * @file
+ * @brief       RIOT header helpers and tools
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#ifdef RIOT_VERSION
+#include "log.h"
+#else
+#include <stdio.h>
+#define LOG_INFO(...) printf(__VA_ARGS__)
+#endif
+
+#include "riot_hdr.h"
+#include "checksum/fletcher32.h"
+
+/**
+ *  @brief Number of header bytes that get checksummed
+ */
+#define RIOT_HDR_CHECKSUM_LEN      (12)
+
+void riot_hdr_print(const riot_hdr_t *riot_hdr)
+{
+    printf("Image magic_number: 0x%08x\n", (unsigned)riot_hdr->magic_number);
+    printf("Image Version: 0x%08x\n", (unsigned)riot_hdr->version);
+    printf("Image start address: 0x%08x\n", (unsigned)riot_hdr->start_addr);
+    printf("Image chksum: 0x%08x\n", (unsigned)riot_hdr->chksum);
```

Comment:
@kYc0o Based on what I understand from the `riot_hdr_t` strcut, this checksum is not the image checksum , but of the checksum of the header (metadata) itself. Am I getting this wrong? Anyhow: there seems to be a contradiction here. Needs clarification.

Typically, the Flash that is checksummed should include not only the metadata but also the image.
But maybe we can take this on in a subsequent PR?

Reply:
True, I can take care of the rewording, but as discussed below, we only aim to have a checksum of the header.

Reply:
@kYc0o ok take care of rewording then